plugins/catalog-backend/src/service/NextRouter.ts

```diff
@@ -37,13 +42,35 @@ import {
 import { disallowReadonlyMode, validateRequestBody } from '../service/util';
 import { RefreshService, RefreshOptions, LocationService } from './types';
 
+const getEntity = async (
+  resourceRef: string,
+  entitiesCatalog: EntitiesCatalog,
+): Promise<Entity | undefined> => {
+  const parsed = parseEntityRef(resourceRef);
+
+  const { entities } = await entitiesCatalog.entities({
+    filter: basicEntityFilter({
+      kind: parsed.kind,
+      'metadata.namespace': parsed.namespace,
+      'metadata.name': parsed.name,
+    }),
+  });
+
+  if (!entities.length) {
+    return undefined;
+  }
+
+  return entities[0];
+};
+
 export interface NextRouterOptions {
   entitiesCatalog?: EntitiesCatalog;
   locationAnalyzer?: LocationAnalyzer;
   locationService: LocationService;
   refreshService?: RefreshService;
   logger: Logger;
   config: Config;
+  permissionRules?: CatalogPermissionRule[];
```

Comment:
I think it's good to leave this optional for now, as catalog permission integration is just beginning, but I'm not sure what the right call is for the long run. On one hand, making it required makes it somewhat explicit that in order for permissions to work, you need to provide some rules. But then you can still provide an empty array even then. On the other hand, leaving it optional feels good because it's not something that should be required if permissions is not enabled at all. With an empty array, you won't get any failures until you try to authorize conditionally, which isn't great, but maybe all this is a moot point - not sure how exposed `createNextRouter` really is in practice?

Reply:
I guess you could say that you're "expected" to use the `CatalogBuilder`.